qiskit/primitives/backend_estimator_dev.py

```diff
@@ -0,0 +1,619 @@
+        self._bound_pass_manager = bound_pass_manager  # TODO: standardize
+        super().__init__(
+            circuits=None,
+            observables=None,
+            parameters=None,
+            options=options,
+        )
+        self._transpile_options = Options()
+
+    ################################################################################
+    ## PROPERTIES
+    ################################################################################
+    @property
+    def backend(self) -> Backend:  # TODO: normalize to one type of Backend
+        """Backend to use for circuit measurements."""
+        return self._backend
+
+    @backend.setter
+    def backend(self, backend: Backend) -> None:
+        if not isinstance(backend, Backend):
+            raise TypeError(
+                "Expected `Backend` type for `backend`, " f"got `{type(backend)}` instead."
+            )
+        # TODO: clear all transpilation caching
+        if isinstance(backend, BackendV1):
+            backend.max_circuits = getattr(backend.configuration(), "max_experiments", None)
+        self._backend = backend
+
+    @property
+    def abelian_grouping(self) -> bool:
+        """Groups commuting observable components."""
+        try:
+            return self._abelian_grouping
+        except AttributeError:
+            return True
+
+    @abelian_grouping.setter
+    def abelian_grouping(self, abelian_grouping: bool) -> None:
+        if not isinstance(abelian_grouping, bool):
+            raise TypeError(
+                "Expected `bool` type for `abelian_grouping`, "
+                f"got `{type(abelian_grouping)}` instead."
+            )
+        self._abelian_grouping = abelian_grouping
+
+    @property
+    def skip_transpilation(self) -> bool:
+        """If `True`, transpilation of the input circuits is skipped."""
+        try:
+            return self._skip_transpilation
+        except AttributeError:
+            return False
+
+    @skip_transpilation.setter
+    def skip_transpilation(self, skip_transpilation: bool) -> None:
+        if not isinstance(skip_transpilation, bool):
+            raise TypeError(
+                "Expected `bool` type for `skip_transpilation`, "
+                f"got `{type(skip_transpilation)}` instead."
+            )
+        self._skip_transpilation = skip_transpilation
+
+    @property
+    def transpile_options(self) -> Options:
+        """Options for transpiling the input circuits."""
+        return self._transpile_options
+
+    def set_transpile_options(self, **fields):
+        """Set the transpiler options for transpiler.
+        Args:
+            **fields: The fields to update the options
+        """
+        self._transpile_options.update_options(**fields)
+
+    ################################################################################
+    ## IMPLEMENTATION
+    ################################################################################
+    # TODO: caching
```

Comment:
These large feature-request todos are likely to stay that way for ever unless it becomes separately clear that there's a need for them.  It's probably best not to put them in at all.

Reply:
Good point, thanks! I plan to go over all the todos before merging, I'll make sure to remove as many as possible 🙂